src/server/network.ts

```diff
@@ -153,8 +154,23 @@ export class Request extends SdkObject {
     return this._headersMap.get(name);
   }
 
-  async rawHeaders(): Promise<NameValue[]> {
-    return this._headers;
+  setWillReceiveExtraHeaders() {
+    if (!this._rawRequestHeadersPromise)
+      this._rawRequestHeadersPromise = new ManualPromise();
+  }
+
+  setRawRequestHeaders(headers: types.HeadersArray) {
+    if (!this._rawRequestHeadersPromise)
+      this._rawRequestHeadersPromise = new ManualPromise();
+    this._rawRequestHeadersPromise!.resolve(headers);
+  }
+
+  async rawRequestHeaders(): Promise<NameValue[]> {
```

Comment:
Please add a comment explaining 3 ways the raw headers could be received

Reply:
Which 3 ways? :)